pkg/imageregistry/imageregistry_test.go

```diff
@@ -0,0 +1,328 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	imageRegistryGVK = schema.GroupVersionKind{
+		Group:   APIGroup,
+		Version: APIVersion,
+		Kind:    APIKind,
+	}
+	defaultImageRegistryName = "cluster"
+	defaultManagementState   = operatorV1.Managed
+)
+
+func TestImageRegistryPull(t *testing.T) {
+	generateImageRegistry := func(name string) *imageregistryV1.Config {
+		return &imageregistryV1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: imageregistryV1.ImageRegistrySpec{
+				OperatorSpec: operatorV1.OperatorSpec{
+					ManagementState: operatorV1.Removed,
+				},
+				Storage: imageregistryV1.ImageRegistryConfigStorage{},
+			},
+		}
+	}
+
+	testCases := []struct {
+		name                string
+		addToRuntimeObjects bool
+		expectedError       error
+		client              bool
+	}{
+		{
+			name:                "test",
+			addToRuntimeObjects: true,
+			expectedError:       nil,
+			client:              true,
+		},
+		{
+			name:                "",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("imageRegistry 'imageRegistryObjName' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                "argocdtest",
+			addToRuntimeObjects: false,
+			expectedError:       fmt.Errorf("imageRegistry object argocdtest doesn't exist"),
+			client:              true,
+		},
+		{
+			name:                "argocdtest",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("imageRegistry Config 'apiClient' cannot be empty"),
+			client:              false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		// Pre-populate the runtime objects
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		testImageRegistry := generateImageRegistry(testCase.name)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testImageRegistry)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects: runtimeObjects,
+			})
+		}
+
+		builderResult, err := Pull(testSettings, testCase.name)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError != nil {
+			assert.NotNil(t, err)
+			assert.Equal(t, testCase.expectedError, err)
+		} else {
+			assert.Nil(t, err)
+			assert.Equal(t, testImageRegistry.Name, builderResult.Object.Name)
+		}
+	}
+}
+
+func TestImageRegistryExist(t *testing.T) {
+	testCases := []struct {
+		testImageRegistryConfig *Builder
+		expectedStatus          bool
+	}{
+		{
+			testImageRegistryConfig: buildValidImageRegistryBuilder(buildImageRegistryClientWithDummyObject()),
+			expectedStatus:          true,
+		},
+		{
+			testImageRegistryConfig: buildInValidImageRegistryBuilder(buildImageRegistryClientWithDummyObject()),
+			expectedStatus:          false,
+		},
+		{
+			testImageRegistryConfig: buildInValidImageRegistryBuilder(clients.GetTestClients(clients.TestClientParams{})),
```

Comment:
Please update it to:
 buildValidImageRegistryBuilder(clients.GetTestClients(clients.TestClientParams{})),